VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -68,10 +82,42 @@
         COURSE
     }
 
+    public static CourseDiscussionPostsThreadFragment newInstance(Bundle args) {
+        CourseDiscussionPostsThreadFragment f = new CourseDiscussionPostsThreadFragment();
+        if (args != null) {
+            f.setArguments(args);
+        }
+        return f;
+    }
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         EventBus.getDefault().register(this);
+
+        if (discussionTopic == null) {
+            // Either we are coming from a deep link or courseware's inline discussion
+            String topicId = getArguments().getString(Router.EXTRA_DISCUSSION_TOPIC_ID);
+            GetSpecificCourseTopicsTask getTopicsTask = new GetSpecificCourseTopicsTask(getContext(),
+                    courseData.getCourse().getId(), Collections.singletonList(topicId)) {
+                @Override
+                protected void onSuccess(CourseTopics courseTopics) throws Exception {
+                    discussionTopic = courseTopics.getCoursewareTopics().get(0).getChildren().get(0);
+                    Activity parent = getActivity();
+                    if (TextUtils.isEmpty(parent.getTitle())) {
+                        // We only need to set the title when it is empty e.g. when coming from a deep link
```

Comment:
We have a placeholder title in `CourseDiscussionPostsActivity` so this condition will never be true. You should instead add an optional argument for a custom Activity title, and only set it to the topic name when that isn't provided, I guess.